patches/minecraft/net/minecraft/world/WorldProvider.java.patch

```diff
@@ -91,39 +93,55 @@
 +    }
 +
 +    /**
-+     * A message to display to the user when they transfer to this dimension.
++     * Gets the message to display to the {@link net.minecraft.entity.Entity} when
++     * they arrive in this dimension.
++     *
++     * <p>This can be a {@link net.minecraft.util.text.TextComponentTranslation tranlatable} or
++     * {@link net.minecraft.util.text.TextComponentString non-translatable} message.</p>
 +     *
 +     * @return The message to be displayed
 +     */
-+    public String getWelcomeMessage()
++    public net.minecraft.util.text.ITextComponent getArrivalMessage()
 +    {
-+        if (this instanceof WorldProviderEnd)
-+        {
-+            return "Entering the End";
-+        }
-+        else if (this instanceof WorldProviderHell)
-+        {
-+            return "Entering the Nether";
-+        }
-+        return null;
++        return this.arrivalMessage;
++    }
++
++    /**
++     * Sets the message to display to the {@link net.minecraft.entity.Entity} when
++     * they arrive in this dimension.
++     *
++     * <p>This can be a {@link net.minecraft.util.text.TextComponentTranslation tranlatable} or
++     * {@link net.minecraft.util.text.TextComponentString non-translatable} message.</p>
++     *
++     * @param arrivalMessage The message to display
++     */
++    public void setArrivalMessage(net.minecraft.util.text.ITextComponent arrivalMessage)
++    {
++        this.arrivalMessage = arrivalMessage;
 +    }
 +
 +    /**
 +     * A Message to display to the user when they transfer out of this dismension.
 +     *
 +     * @return The message to be displayed
 +     */
-+    public String getDepartMessage()
++    public net.minecraft.util.text.ITextComponent getDepartureMessage()
```

Comment:
I've restored the legacy methods, and marked them as deprecated - regenerating patches